include/graphene-box2d.h

```diff
@@ -0,0 +1,122 @@
+ * a maximum vertices lying on the same plane.
+ *
+ * Since: 1.12
+ */
+struct _graphene_box2d_t
+{
+  /*< private >*/
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, min);
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, max);
+};
+
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_alloc                      (void);
+GRAPHENE_AVAILABLE_IN_1_12
+void                      graphene_box2d_free                       (graphene_box2d_t       *box);
+
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init                       (graphene_box2d_t       *box,
+                                                                     const graphene_point_t *min,
+                                                                     const graphene_point_t *max);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_points           (graphene_box2d_t       *box,
+                                                                     unsigned int            n_points,
+                                                                     const graphene_point_t *points);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_vectors          (graphene_box2d_t       *box,
+                                                                     unsigned int            n_vectors,
+                                                                     const graphene_vec2_t  *vectors);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_box              (graphene_box2d_t       *box,
+                                                                     const graphene_box2d_t *src);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_vec2             (graphene_box2d_t       *box,
+                                                                     const graphene_vec2_t  *min,
+                                                                     const graphene_vec2_t  *max);
+
+GRAPHENE_AVAILABLE_IN_1_12
+void                      graphene_box2d_expand                     (const graphene_box2d_t *box,
+                                                                     const graphene_point_t *point,
+                                                                     graphene_box2d_t       *res);
+GRAPHENE_AVAILABLE_IN_1_12
+void                      graphene_box2d_expand_vec2                (const graphene_box2d_t *box,
+                                                                     const graphene_vec2_t  *vec,
+                                                                     graphene_box2d_t       *res);
+GRAPHENE_AVAILABLE_IN_1_12
+void                      graphene_box2d_expand_scalar              (const graphene_box2d_t *box,
+                                                                     float                   scalar,
+                                                                     graphene_box2d_t       *res);
```

Comment:
a linear transform function / fused multiply-add ala `scale * box + offset` is the one thing I've needed in the GSK clip rects to handle affine transforms. And it works as a fallback for all the other transforms (scales, offsets, etc), too.

Reply:
What's the order of operations? Scale first, then offset? Or offset first, and then scale?

Reply:
Another question: should the scale be a single factor, or should be it be decomposed into a vec2?

Basically something like this:

```c
void
graphene_box2d_scale_offset (const graphene_box2d_t *box,
                             const graphene_vec2_t *scale,
                             const graphene_point_t *offset,
                             graphene_box2d_t *res);
```

or with a single `float scale` to be applied to both coordinates?

Reply:
Yeah, that would work. GTK uses 2 different floats for the scale factor everywhere.

(Ideally there'd be a `GRAPHENE_VEC2_INIT_SAME()` macro to easily do the 1 float => 2 floats conversion, but I suppose that's a discussion for a different issue.)